src/com/mendix/widget/carousel/components/Carousel.ts

```diff
@@ -11,16 +13,65 @@ export interface CarouselProps {
     images?: Image[];
 }
 
-const carouselItems = (images: Image[] = []) =>
-    images.map((image, index) => createElement(CarouselItem, {
-        active: index === 0,
-        key: index,
-        url: image.url
-    }));
-
-export const Carousel = (props: CarouselProps) =>
-    DOM.div({ className: "widget-carousel" },
-        DOM.div({ className: "widget-carousel-item-wrapper" },
-            carouselItems(props.images)
-        )
-    );
+interface CarouselState {
+    activeIndex: number;
+}
+
+type Direction = "right" | "left";
+
+export class Carousel extends Component<CarouselProps, CarouselState> {
+    static defaultProps: CarouselProps = { images: [] };
+
+    constructor(props: CarouselProps) {
+        super(props);
+
+        this.state = { activeIndex: 0 };
+    }
+
+    render() {
+        return DOM.div({ className: "widget-carousel" },
+            DOM.div({ className: "widget-carousel-item-wrapper" },
+                this.getCarouselItems(this.props.images, this.state.activeIndex)
+            ),
+            this.props.images.length ? this.getCarouselControls() : null
+        );
+    }
+
+    private getCarouselItems(images: Image[], activeIndex: number) {
```

Comment:
This implies that the carousel items already exist and that you simply are returning them. But you are actually creating them. Why not call it `createCarouselItems` then?